src/schema/notifications.ts

```diff
@@ -145,6 +145,28 @@ export class ZimbraNotifications {
 		this.getApolloClient().queryManager.broadcastQueries();
 	};
 
+	// Find the actual folder of the shared folder
+	private findSharedItemId = (item: any) => {
+		let sharedFolderId: any;
+		const cachedData = get(this.cache, 'data.data');
+		const allFolders = Object.keys(cachedData).filter(f =>
+			f.includes('Folder:')
+		);
+		const idSplit = item.split(':');
+		for (const folderId in allFolders) {
+			const folder: any = cachedData[allFolders[folderId]];
+			//Find the folder where ownerZimbraId:sharedItemId equals to id
+			if (
+				folder.ownerZimbraId === idSplit[0] &&
+				folder.sharedItemId === idSplit[1]
+			) {
+				sharedFolderId = folder.id;
+				break;
```

Comment:
Instead of doing an assignment and a `break` here, you can instead `return folder.id;` and won't need to break out of the loop because the function ends.

Reply:
Done.